### ISSUE TITLE
How  can I use it to work on release X64 Visual Studio machine

### ISSUE THREAD

**Issue by wangwei19900806** [cnr-isti-vclab/vcglib#21]

**Comment by cignoni** [comment]
What do you mean? 
Can you elaborate?

**Comment by wangwei19900806** [comment]
I used the function:"samplePoissonDisk" (vs2013,release,x64) , after I clicked " run" button (visual studio 2013), the visual studio was in a state of "Being generated code" and  it would seem not to proceed.But this is not a problem with the  mode debug, win32

**Comment by wangwei19900806** [comment]
I found the problem only happened in the visual studio with release mode.

**Comment by cignoni** [comment]
Can you show the code where is it happening?

**Comment by wangwei19900806** [comment]
The problem occurred in  "vcg::tri::io::Importer<MyMesh>::Open(m, ply_file_name.c_str())" and the problem is 
solved by using  "vcg::tri::io::ImporterPLY<MyMesh>::Open(m, ply_file_name.c_str())".
Note:m is my defined mesh and ply_file_name is a mesh file.

Thank you!

**Comment by cignoni** [comment]
yes, but without the context (the whole code) it is really hard to understand what is happening to you.

**Comment by afalahat** [comment]
I have the same issue. Visual Studio 2017 Community edition hangs when the project is built in Release mode. It does not have any issue with Debug x86 or x64. The problem is with the Release mode (x86/x64 both does not proceed).

Here is the code I used to decimate a mesh down to 1000 faces. The executable of Debug x86/x64 works just fine.

`	
        

    MyMesh mesh;
	int FinalSize = 1000;
	int err = vcg::tri::io::Importer<MyMesh>::Open(mesh, "Heart_Model.ply");
	if (err)
	{
		printf("Unable to open mesh %s : '%s'\n", "Heart_Model.ply", vcg::tri::io::Importer<MyMesh>::ErrorMsg(err));
		return;
	}

	printf("mesh loaded %d %d \n", mesh.vn, mesh.fn);

	TriEdgeCollapseQuadricParameter qparams;
	qparams.QualityThr = 0.3;
	float TargetError = std::numeric_limits<float>::max();
	bool CleaningFlag = true;

	if (CleaningFlag) {
		int dup = tri::Clean<MyMesh>::RemoveDuplicateVertex(mesh);
		int unref = tri::Clean<MyMesh>::RemoveUnreferencedVertex(mesh);
		printf("Removed %i duplicate and %i unreferenced vertices from mesh \n", dup, unref);
	}


	printf("reducing it to %i\n", FinalSize);

	vcg::tri::UpdateBounding<MyMesh>::Box(mesh);

	// decimator initialization
	vcg::LocalOptimization<MyMesh> DeciSession(mesh, &qparams);

	int t1 = clock();
	DeciSession.Init<MyTriEdgeCollapse>();
	int t2 = clock();
	printf("Initial Heap Size %i\n", int(DeciSession.h.size()));

	DeciSession.SetTargetSimplices(FinalSize);
	DeciSession.SetTimeBudget(0.5f);
	if (TargetError < std::numeric_limits<float>::max()) DeciSession.SetTargetMetric(TargetError);

	while (DeciSession.DoOptimization() && mesh.fn > FinalSize && DeciSession.currMetric < TargetError)
		printf("Current Mesh size %7i heap sz %9i err %9g \r", mesh.fn, int(DeciSession.h.size()), DeciSession.currMetric);

	int t3 = clock();
	printf("mesh  %d %d Error %g \n", mesh.vn, mesh.fn, DeciSession.currMetric);
	printf("\nCompleted in (%5.3f+%5.3f) sec\n", float(t2 - t1) / CLOCKS_PER_SEC, float(t3 - t2) / CLOCKS_PER_SEC);

	vcg::tri::io::ExporterPLY<MyMesh>::Save(mesh, "out2.ply");

`

**Comment by pituke** [comment]
Solution mentioned at https://github.com/cnr-isti-vclab/vcglib/issues/21#issuecomment-294355936 works. I edit _vcglib\vcg\complex\algorithms\create\plymc\plymc.h_ line 183 _tri::io::Importer<SMesh>::Open(m,filename,loadmask)_ to _tri::io::ImporterPLY<SMesh>::Open(m,filename,loadmask)_ and linking successfully ends on Visual Studio 2015 x64 Release.

**Comment by duandongbin000** [comment]
Dear @cignoni @pituke
I have the same problem in VS2017 release x64.  
![image](https://user-images.githubusercontent.com/38930935/84629642-4123d080-af1d-11ea-95ae-c8fe5514828d.png)

**Comment by RuiSheep** [comment]
I solve this problem by setting 
![bug](https://user-images.githubusercontent.com/98319267/165668337-6bcd3028-937a-47a3-ba0c-ddcf3a1ba20e.jpg)
